diff --git a/js/yu.js b/js/yu.js
index 8400c1d..0b24c60 100644
--- a/js/yu.js
+++ b/js/yu.js
@@ -11,7 +11,7 @@
       // Create a fake button to send the video to youtube.
       if ($('.upload-toyoutube').length < 1) {
         $('.field-type-youtube-upload .upload-video')
-            .after('<a href="javascript:;" class="button upload-toyoutube">Upload</a>')
+            .after('<div class="throbber-bootstrap"></div><a href="javascript:;" class="button upload-toyoutube">Upload</a>')
             .end().find('.upload-toyoutube').bind("click", sendToYoutube);
       }
 
@@ -66,7 +66,7 @@
           return false;
         }
         // Attach throbber.
-        $(this).siblings('input[type=submit]').displayThrobber();
+       	$(this).siblings('.throbber-bootstrap').displayThrobber();
         $(this).parents('.form-managed-file').waitingMessage('Authenticating to Youtube...');
 
         var field_ref = $(this).parents('.field-type-youtube-upload').find('input[name$="[youtube_uploader_field_ref]"]').val();
@@ -114,7 +114,8 @@
 
         if (responseText.status != 200) {
           $('form.youtube-upload .form-type-textfield').showErrorMessage('Error code ' + responseText.code + ' status ' + responseText.status);
-          return;
+           $(":submit").attr("disabled", true);
+					return;
         }
         // Add the id to the hidden fid field.
         $('form.youtube-upload .youtube_hidden_id').val(responseText.id);
@@ -124,6 +125,7 @@
         $('.field-type-youtube-upload .form-type-file input').val('');
         // Then trigger the real upload button so Drupal will rebuild everything fine.
         $('.field-type-youtube-upload .upload-video').trigger("mousedown");
+  			$(":submit").removeAttr("disabled");
       }
 
       // Display error message.
